### ISSUE TITLE
Allow to disable/customize focus trap in Dialog

### ISSUE THREAD

**Issue by dzek69** [radix-ui/primitives#2544]
### Overview

I want to be able to disable focus trap when using Dialog component or even better - customize it. Why? I have a Dialog, from which I click the button and now external, vanilla js code kicks in and opens its own dialog over mine. That that extra dialog root element is a sibling to the Radix Dialog, therefore there is no way to focus elements in it, because of focus trap.

### Examples in other libraries

I'm not aware of any other library that provides a dialog with focus trap.

### Who does this impact? Who is this for?

I guess this feature is for people in need of opening multiple dialogs at once. It can happen for anyone randomly, especially when being limited like me by external library.

### Additional context

If you're interested I think I will be able to provide a quick upgrade to allow disabling focus trap. I took a quick look at the code and just disabling it should be quick and easy.

But the best experience would be to customize the trap (so I can allow multiple roots to be focused, but not the "main" page conent). I was thinking about a new prop called `trapFocus` (like you're using internally) that can be either `boolean` or a callback `(target: HTMLElement) => boolean)`. `false` would disable focus trap, `true` would be the current default.
I'd use the callback one like that:

```tsx
<Dialog.Content focusTrap={(target) => {
  return !dialogRootRef.current.contains(target) && !otherModalRef.current.contains(target);
  // return true if target is not contained in known dialogs, disallowing the focus event in this case
 }}>abc</Dialog.Content>
```

The callback one would probably make looping more complex.

**Comment by MananTank** [comment]
Looking at the source code, I found the workaround: 

First, you need to install the `@radix-ui/react-focus-scope` package and then you need to import `FocusScope` and wrap it around the `Dialog.Content` component with `trapped={false}` to stop focus trapping 

```tsx
import { FocusScope } from "@radix-ui/react-focus-scope";

<FocusScope trapped={false}> 
 <Dialog.Content> .... </Dialog.Content>
</FocusScope>

```

**Comment by dzek69** [comment]
@MananTank thanks for your suggestion. But it seems that without blocking outside events Radix dialog will close itself. There is no "sane" way to disable this behavior, see discussion: https://github.com/radix-ui/primitives/discussions/1997

The problem is - if I prevent events outside - then disabling focus trap makes no sense anyway - I still can't focus things in that injected modal from another library.

Also Radix seems to be dead, just as Stitches are :( Which is a shame, as both were very good actually (except the bugs that will now never be fixed).

**Comment by benoitgrelard** [comment]
Duplicate of #1859

**Comment by gruckion** [comment]
Cheers, it's also discussed here.

https://docs.privy.io/guide/react/troubleshooting/ui/dialog

**Comment by ohenrydev** [comment]
Hey guys, I managed to solve this problem for the Dropdown Menu, simply by setting the prop modal explicitly to false

```jsx
	<DropdownMenu modal={false} open={open} onOpenChange={setOpen}>
```

**Comment by SharathVasan** [comment]
> Hey guys, I managed to solve this problem for the Dropdown Menu, simply by setting the prop modal explicitly to false
> 
> ```js
> 	<DropdownMenu modal={false} open={open} onOpenChange={setOpen}>
> ```

This works! I also feel this should be added in the documentation somewhere.

**Comment by Takhine** [comment]
My use-case was to enable animation so I needed to forceMount the Dialog, but it kept stealing the focus on re-render from other inputs

I was able to fix this by doing the following:
```
<RadixDialog.Content onOpenAutoFocus={(e) => e.preventDefault()}>
```
